optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/swingui/ConferenceSchedulingPanel.java

```diff
@@ -16,40 +16,62 @@
 
 package org.optaplanner.examples.conferencescheduling.swingui;
 
+import java.awt.BorderLayout;
 import java.awt.Desktop;
+import java.awt.FlowLayout;
+import java.awt.GridLayout;
 import java.io.File;
 import java.io.IOException;
+
 import javax.swing.JButton;
+import javax.swing.JComboBox;
 import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
 
+import org.optaplanner.examples.common.business.SolutionBusiness;
 import org.optaplanner.examples.common.swingui.SolutionPanel;
 import org.optaplanner.examples.conferencescheduling.domain.ConferenceSolution;
 import org.optaplanner.examples.conferencescheduling.persistence.ConferenceSchedulingCfpDevoxxImporter;
 import org.optaplanner.examples.conferencescheduling.persistence.ConferenceSchedulingXlsxFileIO;
 import org.optaplanner.persistence.common.api.domain.solution.SolutionFileIO;
 
+import static java.util.concurrent.Executors.newSingleThreadExecutor;
+
 public class ConferenceSchedulingPanel extends SolutionPanel<ConferenceSolution> {
 
     public static final String LOGO_PATH = "/org/optaplanner/examples/conferencescheduling/swingui/conferenceSchedulingLogo.png";
 
     public ConferenceSchedulingPanel() {
-        JButton importConferenceButton = new JButton("Import conference");
+        JButton importConferenceButton = new JButton("Import from CFP");
         importConferenceButton.addActionListener(event -> {
-            //TODO: Add a panel to get conferenceBaseUrl from the user
-            ConferenceSchedulingCfpDevoxxImporter conferenceSchedulingImporter = new ConferenceSchedulingCfpDevoxxImporter("https://dvbe18.confinabox.com/api/conferences/DVBE18");
-            solutionBusiness.setSolution(conferenceSchedulingImporter.importSolution());
+            String[] cfpArray = {"devoxx-cfp"};
+            JComboBox cfpConferenceBox = new JComboBox(cfpArray);
+            JTextField cfpRestUrlTextField = new JTextField("https://dvbe18.confinabox.com/api/conferences/DVBE18");
+            Object[] dialogue = {
+                    "Choose conference:", cfpConferenceBox,
+                    "Enter CFP REST Url:", cfpRestUrlTextField
+            };
+
+            int option = JOptionPane.showConfirmDialog(this, dialogue, "Import", JOptionPane.OK_CANCEL_OPTION);
+            if (option == JOptionPane.OK_OPTION) {
+                String conferenceBaseUrl = cfpRestUrlTextField.getText();
+                newSingleThreadExecutor().execute(new ConferenceImporter(conferenceBaseUrl, solutionBusiness));
```

Comment:
ExecutorService needs to be shutdown to properly clean up running threads. Please see https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html for details.

Reply:
It looks too much complication for executing one task here, is there a reason not to use new Thread().start() instead?

Reply:
@MusaTalluzi I am not aware of any, new Thread().start() should work too.

Reply:
Don't use new Thread().start() but use SwingWorker (which wraps that) instead.
See SolveWorker in SolverAndPersistenceFrame